hail/src/main/scala/is/hail/expr/ir/InferPType.scala

```diff
@@ -24,6 +24,7 @@ object InferPType {
       case Cast(_, t) => PType.canonical(t)
       case NA(t) => PType.canonical(t)
       case IsNA(_) => PBoolean()
+      case Coalesce(values) => values.head.pType // FIXME this will be wrong
```

Comment:
Can you clarify on the `FIXME`?

Reply:
I'm just going to delete InferPType until we actually have ptypes, I think.

This will be wrong when we have actual ptypes because the values could have different physical types, requiring us to unify them or use a union type.

Reply:
Gotcha, sounds good